### PR TITLE
Add required `permanent: boolean` field to redirect

### PR DESCRIPTION
**Pull request by ijjk**
As discussed this requires a `permanent: true/false` value be specified if a `statusCode` isn't. This helps prevent us from inaccurately assuming a default redirect status for the user. 

Allowing the user to tell us whether it's `permanent` or not creates a safer config than having them try to figure out the correct status code to use

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| buildDuration | 13.5s | 13.5s | ⚠️ +23ms |
| nodeModulesSize | 49 MB | 49 MB | ⚠️ +964 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.11 kB | 5.11 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..9881.js gzip | 13.7 kB | 13.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 67.4 kB | 67.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.17 kB | 4.17 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 12.5 kB | 12.5 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 62.1 kB | 62.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.js gzip | 1.33 kB | 1.33 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.9 kB | 2.9 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.87 kB | 9.87 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.module.js gzip | 757 B | 757 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.47 kB | 2.47 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.41 kB | 7.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| index.html gzip | 1.02 kB | 1.02 kB | ✓ |
| link.html gzip | 1.03 kB | 1.03 kB | ✓ |
| withRouter.html gzip | 1.02 kB | 1.02 kB | ✓ |
| Overall change | 3.07 kB | 3.07 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| buildDuration | 14.1s | 14s | -130ms |
| nodeModulesSize | 49 MB | 49 MB | ⚠️ +964 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.11 kB | 5.11 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..9881.js gzip | 13.7 kB | 13.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 67.4 kB | 67.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.17 kB | 4.17 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 12.5 kB | 12.5 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 62.1 kB | 62.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.js gzip | 1.33 kB | 1.33 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.9 kB | 2.9 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.87 kB | 9.87 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.module.js gzip | 757 B | 757 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.47 kB | 2.47 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.41 kB | 7.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _error.js gzip | 77.8 kB | 77.8 kB | ⚠️ +13 B |
| hooks.html gzip | 1.05 kB | 1.05 kB | ✓ |
| index.js gzip | 78 kB | 78 kB | ⚠️ +13 B |
| link.js gzip | 80.4 kB | 80.4 kB | ⚠️ +13 B |
| routerDirect.js gzip | 78.1 kB | 78.1 kB | ⚠️ +14 B |
| withRouter.js gzip | 78.1 kB | 78.1 kB | ⚠️ +14 B |
| Overall change | 393 kB | 393 kB | ⚠️ +67 B |

</details>

</details>


Commit: 5adbf6b3b53cf49700e46ccad88804c0b8983917

File: test/integration/custom-routes/test/index.test.js [hunk 1]
```diff
@@ -278,13 +278,15 @@ const runTests = (isDev = false) => {
             statusCode: 307,
             regex: normalizeRegEx('^\\/hello(?:\\/([^\\/]+?))\\/another$'),
             regexKeys: ['id'],
+            permanent: false,
```
Comment (timneutkens) [inline test/integration/custom-routes/test/index.test.js]: Isn't this not needed because statusCode is generated by the framework.

Comment (ijjk) [reply test/integration/custom-routes/test/index.test.js]: Right we shouldn't need it, updated to not output it

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| buildDuration | 12.7s | 12.8s | ⚠️ +99ms |
| nodeModulesSize | 49 MB | 49 MB | ⚠️ +1.02 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.11 kB | 5.11 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..9881.js gzip | 13.7 kB | 13.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 67.5 kB | 67.5 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.18 kB | 4.18 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 12.5 kB | 12.5 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 62.1 kB | 62.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.js gzip | 1.33 kB | 1.33 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.9 kB | 2.9 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.87 kB | 9.87 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.module.js gzip | 757 B | 757 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.47 kB | 2.47 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.41 kB | 7.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| index.html gzip | 1.02 kB | 1.02 kB | ✓ |
| link.html gzip | 1.03 kB | 1.03 kB | ✓ |
| withRouter.html gzip | 1.01 kB | 1.01 kB | ✓ |
| Overall change | 3.06 kB | 3.06 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| buildDuration | 13.3s | 13.2s | -135ms |
| nodeModulesSize | 49 MB | 49 MB | ⚠️ +1.02 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.11 kB | 5.11 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..9881.js gzip | 13.7 kB | 13.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 67.5 kB | 67.5 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.18 kB | 4.18 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 12.5 kB | 12.5 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 62.1 kB | 62.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.js gzip | 1.33 kB | 1.33 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.9 kB | 2.9 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.87 kB | 9.87 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.module.js gzip | 757 B | 757 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.47 kB | 2.47 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.41 kB | 7.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _error.js gzip | 77.8 kB | 77.8 kB | ⚠️ +14 B |
| hooks.html gzip | 1.05 kB | 1.05 kB | ✓ |
| index.js gzip | 78 kB | 78 kB | ⚠️ +13 B |
| link.js gzip | 80.4 kB | 80.4 kB | ⚠️ +13 B |
| routerDirect.js gzip | 78.1 kB | 78.1 kB | ⚠️ +14 B |
| withRouter.js gzip | 78.1 kB | 78.1 kB | ⚠️ +13 B |
| Overall change | 393 kB | 393 kB | ⚠️ +67 B |

</details>

</details>


Commit: b97a6447d97653e123ed2f0fe6d030271b0f9133

**Review by Timer — APPROVED** [review]
LGTM

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| buildDuration | 13.3s | 13.1s | -117ms |
| nodeModulesSize | 49 MB | 49 MB | ⚠️ +1.02 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.11 kB | 5.11 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..9881.js gzip | 13.7 kB | 13.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 67.5 kB | 67.5 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.18 kB | 4.18 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 12.5 kB | 12.5 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 62.1 kB | 62.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.js gzip | 1.33 kB | 1.33 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.9 kB | 2.9 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.87 kB | 9.87 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.module.js gzip | 757 B | 757 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.47 kB | 2.47 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.41 kB | 7.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| index.html gzip | 1.02 kB | 1.02 kB | ✓ |
| link.html gzip | 1.03 kB | 1.03 kB | ✓ |
| withRouter.html gzip | 1.01 kB | 1.01 kB | ✓ |
| Overall change | 3.06 kB | 3.06 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| buildDuration | 13.8s | 13.8s | ⚠️ +29ms |
| nodeModulesSize | 49 MB | 49 MB | ⚠️ +1.02 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.js gzip | 5.11 kB | 5.11 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..54d3.js gzip | 4.68 kB | 4.68 kB | ✓ |
| commons.HASH.js gzip | 4.06 kB | 4.06 kB | ✓ |
| de003c3a9d30..9881.js gzip | 13.7 kB | 13.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 67.5 kB | 67.5 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 4.18 kB | 4.18 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 4952ddcd88e7..dule.js gzip | 5.56 kB | 5.56 kB | ✓ |
| de003c3a9d30..dule.js gzip | 12.5 kB | 12.5 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 62.1 kB | 62.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 4.76 kB | 4.76 kB | ✓ |
| Overall change | 4.76 kB | 4.76 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.js gzip | 1.33 kB | 1.33 kB | ✓ |
| _error.js gzip | 4.07 kB | 4.07 kB | ✓ |
| hooks.js gzip | 779 B | 779 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.9 kB | 2.9 kB | ✓ |
| routerDirect.js gzip | 283 B | 283 B | ✓ |
| withRouter.js gzip | 282 B | 282 B | ✓ |
| Overall change | 9.87 kB | 9.87 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _app.module.js gzip | 757 B | 757 B | ✓ |
| _error.module.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js gzip | 2.47 kB | 2.47 kB | ✓ |
| routerDirect..dule.js gzip | 273 B | 273 B | ✓ |
| withRouter.m..dule.js gzip | 272 B | 272 B | ✓ |
| Overall change | 7.41 kB | 7.41 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _buildManifest.js gzip | 61 B | 61 B | ✓ |
| _buildManife..dule.js gzip | 61 B | 61 B | ✓ |
| Overall change | 122 B | 122 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary  | ijjk/next.js add/permanent-redirect-field | Change |
| - | - | - | - |
| _error.js gzip | 77.8 kB | 77.8 kB | ⚠️ +14 B |
| hooks.html gzip | 1.05 kB | 1.05 kB | ✓ |
| index.js gzip | 78 kB | 78 kB | ⚠️ +13 B |
| link.js gzip | 80.4 kB | 80.4 kB | ⚠️ +13 B |
| routerDirect.js gzip | 78.1 kB | 78.1 kB | ⚠️ +14 B |
| withRouter.js gzip | 78.1 kB | 78.1 kB | ⚠️ +13 B |
| Overall change | 393 kB | 393 kB | ⚠️ +67 B |

</details>

</details>


Commit: 8127db202a7916fba8ad2cf0684d52c5442ec53a

**Pull request merged**